src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -611,25 +612,55 @@ class spell_warr_shockwave : public SpellScript
     uint32 _targetCount = 0;
 };
 
-// 107570 - Storm Bolt
+// Warrior: Arms, Fury, Protection
+// Storm Bolt (SpellID: 107570)
 class spell_warr_storm_bolt : public SpellScript
 {
     bool Validate(SpellInfo const* /*spellInfo*/) override
     {
-        return ValidateSpellInfo
-        ({
-            SPELL_WARRIOR_STORM_BOLT_STUN
+        return ValidateSpellInfo({
+            SPELL_WARRIOR_STORM_BOLT_STUN,
+            SPELL_WARRIOR_STORM_BOLTS
         });
     }
 
-    void HandleOnHit(SpellEffIndex /*effIndex*/)
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        Unit* caster = GetCaster();
+
+        if (!caster)
+            return;
+
+        // If the caster has the Storm Bolts aura, do not filter targets
+        if (caster && caster->HasAura(SPELL_WARRIOR_STORM_BOLTS))
+            return;
+
+        // Clear the current targets
+        targets.clear();
+
+        // Add the desired target back to the list
+        if (Unit* target = GetExplTargetUnit())
+        {
+            targets.push_back(GetExplTargetUnit());
+        }
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
     {
-        GetCaster()->CastSpell(GetHitUnit(), SPELL_WARRIOR_STORM_BOLT_STUN, true);
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+
+        if (!caster || !target)
```

Comment:
this is not necessary as caster and target are guaranteed to exist on OnEffectHitTarget